anomdetect/src/main/java/com/expedia/adaptivealerting/anomdetect/PerformanceMonitor.java

```diff
@@ -34,37 +35,46 @@
     private int tickCounter;
 
     /**
-     * Local RMSE evaluator.
+     * Local evaluator.
      */
-    private RmseEvaluator evaluator;
+    private Evaluator evaluator;
 
     /**
-     * Local threshold lookup where we maintain thresholds for different evaluators.
+     * Local performance monitor listener.
      */
+    private PerfMonListener listener;
 
-    public static final int MAX_TICKS = 100;
+    /**
+     * Local max no ticks where performance monitor resets evaluator.
+     */
+
+    private int maxTicks;
 
     /**
      * Creates a new performance monitor and sets evaluator as RMSE evaluator
      */
-    public PerformanceMonitor() {
+    public PerformanceMonitor(PerfMonListener listener, Evaluator evaluator, int maxTicks) {
+        this.listener = listener;
```

Comment:
Please validate the params here, as this is an API method.. Neither listener nor evaluator should be allowed to be null, and maxTicks should be > 0.